src/components/HomepageFeatures/animation.tsx

```diff
@@ -0,0 +1,216 @@
+      let moveX = 0
+      if (scaledT < 0.25) {
+        moveX = Utils.map(scaledT, 0, 0.25, 0, config.size)
+        moveY = Utils.map(scaledT, 0, 0.25, 0, 0)
+      } else if (scaledT >= 0.25 && scaledT < 0.5) {
+        moveX = Utils.map(scaledT, 0.25, 0.5, config.size, config.size)
+        moveY = Utils.map(scaledT, 0.25, 0.5, 0, config.size)
+      } else if (scaledT >= 0.5 && scaledT < 0.75) {
+        moveX = Utils.map(scaledT, 0.5, 0.75, config.size, 0)
+        moveY = Utils.map(scaledT, 0.5, 0.75, config.size, config.size)
+      } else {
+        moveX = Utils.map(scaledT, 0.75, 1.0, 0, 0)
+        moveY = Utils.map(scaledT, 0.75, 1.0, config.size, 0)
+      }
+
+      ctx.fillStyle = color
+      ctx.beginPath()
+      ctx.arc(
+        config.shapes[i].x + moveX,
+        config.shapes[i].y + moveY,
+        config.size / 3 
+        + j,
+        0,
+        Math.PI * 2,
+        false
+      )
+      ctx.fill()
+      ctx.restore()
+    }
+  }
+  ctx.restore()
+  return t
+}
+
+class Utils {
+  static norm(value, min, max) {
+    return (value - min) / (max - min)
+  }
+
+  static lerp(norm, min, max) {
+    return (max - min) * norm + min
+  }
+
+  static map(value, sourceMin, sourceMax, destMin, destMax) {
+    return this.lerp(this.norm(value, sourceMin, sourceMax), destMin, destMax)
+  }
+
+  static clamp(value, min, max) {
+    return Math.min(Math.max(value, min), max)
+  }
+
+  static distance(x0, y0, x1, y1) {
+    const dx = x1 - x0
+    const dy = y1 - y0
+
+    return Math.sqrt(dx * dx + dy * dy)
+  }
+
+  static randomRange(min, max) {
+    return min + Math.random() * (max - min)
+  }
+
+  static randomInt(min, max) {
+    return Math.floor(min + Math.random() * (max - min + 1))
+  }
+
+  static randomDist(min, max, iterations) {
+    let total = 0
+
+    for (let i = 0; i < iterations; i++) {
+      total += this.randomRange(min, max)
+    }
+
+    return total / iterations
+  }
+
+  static degreesToRads(degrees) {
+    return (degrees / 180) * Math.PI
+  }
+
+  static radsToDegrees(radians) {
+    return (radians * 180) / Math.PI
+  }
+
+  static roundToPlaces(value, places) {
+    const mult = Math.pow(10, places)
+
+    return Math.round(value * mult) / mult
+  }
+
+  static roundNearest(value, nearest) {
+    return Math.round(value / nearest) * nearest
+  }
+}
+
+export default function  Animation({ className: string }): JSX.Element {
+  const canvasRef = useRef(null)
+
+  useEffect(() => {
+    const canvas = canvasRef.current
+    const context = canvas.getContext('2d')
+
+    const config = initialize(canvas)
+    let animationFrameId: number
+    const render = (timestamp: number) => {
+      draw(config, context, timestamp)
+      animationFrameId = window.requestAnimationFrame(render)
+    }
+    render(0)
+
+    return () => {
+      window.cancelAnimationFrame(animationFrameId)
+    }
+  }, [draw])
+
+  return <canvas className={"animation"} ref={canvasRef} width="600" height="600" style={{}} />
```

Comment:
For strings trengs ikke objekter. Tom style har heller ingen effekt
```suggestion
  return <canvas className="animation" ref={canvasRef} width="600" height="600" />
```